src/libraries/Microsoft.PowerFx.Core/Binding/BinderExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Binding
+{
+    internal static class BinderExtensions
+    {
+        public static void AddIfNotSameSpan(this IList<KeyValuePair<Token, string>> nodes, Token key, string value)            
+        {
+            foreach (KeyValuePair<Token, string> kvp in nodes)
```

Comment:
1. This introduces a O[N^2]
2. We should understand _why_ are we adding duplicates? That maybe the real bug.   
This switches to a "first add wins" policy. If the first and second are adding different values, this could pick the wrong one - when the real bug may be that we shouldn't be adding twice at all.